### ISSUE TITLE
[JOSS] Using modified versions of pvfactors and bifacial_radiance seems unsustainable

### ISSUE THREAD

**Issue by kanderso-nrel** [cire-thk/BifacialSimu#8]
The installation instructions in the README say to install pvfactors and bifacial_radiance, then edit those installations to use modified files in this repository.  This approach seems problematic to me for a few reasons:

- It makes installation more manual and error-prone.  
- It is brittle: the original projects will continue to evolve and diverge from the old code in this repository.  Maybe someday they will diverge enough that your modified code becomes incompatible with the latest version of pvfactors or bifacial_radiance and the code breaks.
- By replacing their code with yours, it means any improvements they make to those packages might not apply to your code.  For example maybe someday someone finds a bug in pvfactors's `pvground.py` and fixes it.  BifacialSimu won't get the bugfix because it is overwriting that file with an old version from before the bugfix.
- It isn't aligned with the open-source ethos of sharing improvements with the wider community.  Ideally, any improvements you make to a dependency would be contributed to the dependency itself so that everyone can benefit from it. 

If you compare the bifacial_radiance `main.py` in this repository with the latest version in their repository ([here](https://github.com/NREL/bifacial_radiance/blob/development/bifacial_radiance/main.py)), there are already many differences.  I am worried that using modified files will not be sustainable in the long term, and maybe not even in the short term.  

Would it be possible to either get rid of those changes, or to contribute those changes back to the original projects?  Maybe there is some other solution to this problem?

Cross-reference: https://github.com/openjournals/joss-reviews/issues/4443

**Comment by EwaGomez** [comment]
Thank you for your suggestion! I am already working on it. Unfortunatly the file changes are nesceary right now to apply our albedo models. I totally agree that it makes the installation more manual. My students and I will try to find a solution, it was already on our To-Do list, but I thought I would publish the software beforehand to make it open source and to get more suggestions. Thank you!

**Comment by EwaGomez** [comment]
Dear @kanderso-nrel and @sidihamady ,

Unfortunately, the changes in pyground.py and main.py are necessary. My idea now is to implement copies of pvfactors and bifacial_radiance into BifacialSimu to integrate them into the pip-installation process. In those copies, the necessary changes are already done, so users do not have to exchange files themselves, which would prevent errors.

Doing so, when there is an update in bifacial_radiance or pvfactors, I could transfer my changes to the newer versions and test them. In the long run, I would like to contribute my changes to bifacial_radiance and pvfactors, but for now, I think that would be a good solution. 

Is that suitable for you?

**Comment by kanderso-nrel** [comment]
I think that would address the first and second bullets in my list above.  Perhaps it would be helpful to open an issue in each repository describing your changes?  Maybe there is a better solution that the maintainers of those packages could suggest, and they could also indicate whether they would be interested in a PR for those changes, even if it will happen someday instead of today.

One other thing is I think you would have to also start tracking all of the dependencies of pvfactors and bifacial_radiance in addition to your own dependencies (#11).  FYI this practice of embedding one project within another is sometimes called "vendoring", in case you want to search for advice on this topic.

**Comment by EwaGomez** [comment]
Thank you for your advice with the vendoring! We did as you said and made the installation process a lot easier using a setup.py and pip install . The changes are also integrated into the ReadMe file as well as the requirements. Doing so, we changed the vendored version of bifacial_radiance to 0.4.1.

I will definitely open an issue in each repository with my changes and ask them for advice. Thank you!
